src/valiant/models/cdna_seq_repository.py

```diff
@@ -0,0 +1,114 @@
+# modification, transmission or transference is done without financial return, the conditions of this Licence are imposed
+# upon the receiver of the product, and all original and amended source code is included in any transmitted product. You
+# may be held legally responsible for any copyright infringement that is caused or encouraged by your failure to abide by
+# these terms and conditions.
+#
+# You are not permitted under this Licence to use this Software commercially. Use for which any financial return is
+# received shall be defined as commercial use, and includes (1) integration of all or part of the source code or the
+# Software into a product for sale or license by or on behalf of Licensee to third parties or (2) use of the Software
+# or any derivative of it for research with the final aim of developing software products for sale or license to a third
+# party or (3) use of the Software or any derivative of it for research with the final aim of developing non-software
+# products for sale or license to a third party, or (4) use of the Software to provide any service to an external
+# organisation for which payment is received. If you are interested in using the Software commercially, please contact
+# legal@sanger.ac.uk. Contact details are: legal@sanger.ac.uk quoting reference Valiant-software.
+#############################
+
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Dict, Optional, Iterable, FrozenSet, List, Tuple
+from ..loaders.fasta import load_from_multi_fasta
+from ..loaders.tsv import load_tsv
+from .base import PositionRange, StrandedPositionRange
+from .cdna import CDNA, AnnotatedCDNA
+from .sequences import Sequence
+from .sequence_info import SequenceInfo
+
+
+CDNA_ANNOT_FIELDS = [
+    'seq_id',
+    'gene_id',
+    'transcript_id',
+    'cds_start',
+    'cds_end'
+]
+
+
+def load_seqs(fp: str, ids: Iterable[str]) -> Dict[str, Sequence]:
+    return {
+        seq_id: Sequence(seq)
+        for seq_id, seq in load_from_multi_fasta(fp, ids).items()
+    }
+
+
+def load_annot(fp: str, ids: Iterable[str]) -> Dict[str, Tuple[SequenceInfo, Optional[StrandedPositionRange]]]:
+    def parse_cds_range(cds_start: str, cds_end: str) -> Optional[StrandedPositionRange]:
+        if bool(cds_start) != bool(cds_end):
+            raise ValueError("Invalid CDS range!")
+        if not cds_start:
+            return None
+        return StrandedPositionRange(
+            int(cds_start), int(cds_end), '+')
+
+    return {
+        seq_id: (
+            SequenceInfo(gene_id or None, transcript_id or None),
+            parse_cds_range(cds_start, cds_end)
+        )
+        for seq_id, gene_id, transcript_id, cds_start, cds_end in load_tsv(
+            fp, CDNA_ANNOT_FIELDS)
+        if seq_id in ids
+    }
+
+
+# TODO: currently supporting only all-or-none cDNA annotation...?
```

Comment:
Possibly, yes; it will stay as it is for now as the behaviour in case of mixed annotation statuses may be difficult to communicate to the user (who may have just forgotten provide annotation for some of the targets, and this may result in unexpected errors due to the requirements of some mutation patterns).